staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -141,22 +141,16 @@ func WithMaxInFlightLimit(
 
 			select {
 			case c <- true:
-				var mutatingLen, readOnlyLen int
+				defer func() {
+					<-c
+				}()
+
 				if isMutatingRequest {
-					mutatingLen = len(mutatingChan)
+					watermark.recordMutating(len(mutatingChan))
 				} else {
-					readOnlyLen = len(nonMutatingChan)
+					watermark.recordReadOnly(len(nonMutatingChan))
```

Comment:
I'd prefer to count this before `select` in order to catch the case of exceeding the limit.

Reply:
We can only count the exact total inflight requests in `select`. Could you please inform what limit will be exceeded?

Reply:
I mean, for now, we only count in `case c <- true:`, how about the request goes to `default:`? 

Considering that the gauge metric will be refresh after 1s, why not refresh the current inflight count when each request comes?

Reply:
`default` case indicates that the inflight queue is full and apiserver will not futher handle the incoming requests with `too many requests` returned. So, dropped requests should not be considered as `inflight` requests.